src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
 import './PostsList.scss';
+import classNames from 'classnames';
+import { getPosts, getUserPosts } from '../../api/posts';
 
-export const PostsList = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+export const PostsList = ({ selectedUser, onChangePost, selectedPost }) => {
+  const [posts, setPosts] = useState([]);
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+  useEffect(() => {
+    if (selectedUser === 0) {
+      getPosts().then(data => setPosts(data));
+    } else {
+      getUserPosts(selectedUser).then(data => setPosts(data));
+    }
+  }, [selectedUser]);
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
+  const postHandler = (postId) => {
+    if (postId === selectedPost) {
+      onChangePost(null);
+    } else {
+      onChangePost(postId);
+    }
+  };
 
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="PostsList">
+      <h2>Posts:</h2>
+
+      <ul className="PostsList__list">
+        {posts.length ? (
+          posts.map(({ id, userId, title, body }) => (
+            <li key={id} className="PostsList__item">
+              <div>
+                <b>
+                  [User #$
+                  {userId}
+                  ]:
+                  {' '}
+                </b>
+                {title}
+              </div>
+              <button
+                type="button"
+                className={classNames(
+                  'PostsList__button button',
+                  { active: id === selectedPost },
+                )}
+                onClick={() => postHandler(id)}
+              >
+                {selectedPost === id ? 'Close' : 'Open'}
+              </button>
+            </li>
+          ))
+        ) : (
+          <p>User have not post</p>
+        )
+        }
```

Comment:
```suggestion
        ) : (
          <p>User have not post</p>
        )}
```
И опять короткий случай после полотна